### PR TITLE
fix(agent): support custom interruptible model hooks

### LINKED ISSUE #4091: [Question] HumanInTheLoop 没法扩展么？
### Question

`		// Add hook nodes for afterModel hooks
		for (Hook hook : afterModelHooks) {
			if (hook instanceof ModelHook modelHook) {
				if (hook instanceof HumanInTheLoopHook humanInTheLoopHook) {
					graph.addNode(Hook.getFullHookName(hook) + ".afterModel", humanInTheLoopHook);
				} else {
					graph.addNode(Hook.getFullHookName(hook) + ".afterModel", modelHook::afterModel);
				}
			} else if (hook instanceof MessagesModelHook messagesModelHook) {
				graph.addNode(Hook.getFullHookName(hook) + ".afterModel", MessagesModelHook.afterModelAction(messagesModelHook));
			}
		}`
ReactAgent的人机协作这里，完全写死了啊，我都没法自定义一个其他能中断的hook，人机协作又不是只有工具同意拒绝这种

### PR DESCRIPTION
**Pull request by March-77**
## Summary

- preserve custom model hook instances when they implement both `AsyncNodeActionWithConfig` and `InterruptableAction`
- apply the same capability-based registration to before-model and after-model hooks
- add an end-to-end regression test proving a custom after-model hook interrupts a real ReactAgent execution

## Root cause

ReactAgent registered only the concrete built-in `InterruptionHook` and `HumanInTheLoopHook` instances as graph node actions. Every other `ModelHook` was converted to a method reference such as `modelHook::afterModel`. That wrapper no longer implemented `InterruptableAction`, so GraphRunner could not invoke a custom hook's interruption contract even when the original hook implemented the existing public interfaces.

The fix uses those existing capabilities instead of concrete class names and does not add or change any public API.

## Validation

- `./mvnw -pl spring-ai-alibaba-agent-framework -Dtest=CustomInterruptibleHookTest,InterruptionHookTest,HumanInTheLoopTest test` — 21 tests, 0 failures, 8 environment-gated skips
- `./mvnw -pl spring-ai-alibaba-agent-framework test` — 591 tests, 0 failures, 167 environment-gated skips
- Checkstyle and Spotless passed

Fixes #4091

### CODE REVIEW

File: spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/ReactAgent.java [hunk 1]
```diff
@@ -369,8 +369,8 @@ protected StateGraph initGraph() throws GraphStateException {
 		// Add hook nodes for beforeModel hooks
 		for (Hook hook : beforeModelHooks) {
 			if (hook instanceof ModelHook modelHook) {
-				if (hook instanceof InterruptionHook interruptionHook) {
-					graph.addNode(Hook.getFullHookName(hook) + ".beforeModel", interruptionHook);
+				if (hook instanceof AsyncNodeActionWithConfig action && hook instanceof InterruptableAction) {
+					graph.addNode(Hook.getFullHookName(hook) + ".beforeModel", action);
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/ReactAgent.java]: Fixed in `fe2ac8173`. Interruptible model hooks now wrap the position-specific `beforeModel`/`afterModel` callback while delegating both interruption phases. The regression test registers one hook at both positions, verifies both callbacks run exactly once, and verifies its generic `apply` is never used. `CustomInterruptibleHookTest` passes; module Checkstyle reports 0 violations.

File: spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/ReactAgent.java [hunk 2]
```diff
@@ -369,8 +371,9 @@ protected StateGraph initGraph() throws GraphStateException {
 		// Add hook nodes for beforeModel hooks
 		for (Hook hook : beforeModelHooks) {
 			if (hook instanceof ModelHook modelHook) {
-				if (hook instanceof InterruptionHook interruptionHook) {
-					graph.addNode(Hook.getFullHookName(hook) + ".beforeModel", interruptionHook);
+				if (hook instanceof InterruptableAction interruptableAction) {
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/ReactAgent.java:380]: Fixed in `78d69e767`. The position action is now selected for either `ModelHook` or `MessagesModelHook` before the shared interruptible wrapper is applied. The new regression runs an interruptible MessagesModelHook through its after-model callback and verifies `interruptAfter` produces the interruption; both hook lifecycle tests pass.

**Review by oss-sentinel-ai — COMMENTED** [review]
### Summary
Reviewed the changes in this PR (339 lines changed). The changes appear to be well-structured.

### Observations
- Changes span 339 lines
- Please ensure all edge cases are covered
- Verify test coverage for the modified code paths

---
*Automated review by github-manager-bot*

**Comment by March-77** [comment]
@yuluo-yx 已重新核对当前 head、最新 main 变化、mergeability、CI、reviews 和 review threads；现有评审意见均已验证并清理，未发现需要追加代码修改的问题。烦请在方便时进行维护者审查，谢谢。
